### ISSUE TITLE
Renderer: potential buffer overrun if not enough space for swap buffers command

### ISSUE THREAD

**Issue by xycaleth** [ioquake/ioq3#147]
The following comes from `code/renderergl1/tr_cmds.c` in `RE_EndFrame`:
```c
	cmd = R_GetCommandBuffer( sizeof( *cmd ) );
	if ( !cmd ) {
		return;
	}
	cmd->commandId = RC_SWAP_BUFFERS;

	R_IssueRenderCommands( qtrue );
```
If `R_GetCommandBuffer` returns `NULL`, then the swap buffers command cannot be added to the command buffer, and `RE_EndFrame` returns early. As the render commands are never executed (and the command buffer size is not reset), the `tr.refdef.drawSurfs` array is continually appended to every "frame", until `tr.refdef.numDrawSurfs` exceeds `MAX_DRAW_SURFS` and the game potentially crashes.

`R_IssueRenderCommands` looks like it ignores whether swap buffers happens and simply appends the end of list command, executes the commands, and then resets the command buffer. Even if this was called early to reset the command buffer, it would still be an issue because none of the subsequent frames are swapped and the player would just see the last frame that was swapped.

**Comment by zturtleman** [comment]
Yeah, I'd run into this when drawing more console text. Though, my solution for it might mess up r_speeds. https://gist.github.com/zturtleman/1a16197add4e4234432e#file-qq_reflow_console-diff-L132

**Comment by zturtleman** [comment]
Fixed in 8531162bd9761ce49016b3738ddf394b28f8e462, e6209f3b7c008d6d6ec933788088510ad04cd2c5, 81e2b6c0cf1bf8f7cea18035d24edb1ff8ae36ff.
